code/game/objects/items/holosign_creator.dm

```diff
@@ -64,23 +67,33 @@
 		if(target_turf.is_blocked_turf(TRUE)) //don't try to sneak dense stuff on our tile during the wait.
 			return .
 	target_holosign = new holosign_type(get_turf(target), src)
+	if(color)
+		target_holosign.color = color
 	return .
 
 /obj/item/holosign_creator/attack(mob/living/carbon/human/M, mob/user)
 	return
 
 /obj/item/holosign_creator/attack_self(mob/user)
 	if(LAZYLEN(signs))
-		for(var/H in signs)
-			qdel(H)
+		for(var/obj/structure/holosign/hologram as anything in signs)
```

Comment:
Shouldn't these be using ``as anything`` since we know what the lists should have?

Reply:
yes, I changed them to anything. though I missed a couple there.